File: contests/abc352/abc352_a.ml
open Core
open Scanf

let contains ~lower ~upper x = lower <= x && x <= upper
let cond b = if b then "Yes" else "No"

let () =
  scanf "%d %d %d %d" @@ fun n x y z ->
  contains z ~lower:(min x y) ~upper:(max x y) |> cond |> print_endline

File: contests/abc352/abc352_a.hs
contains :: (Int, Int) -> Int -> Bool
(l, u) `contains` n = l <= n && n <= u

main :: IO ()
main = do
  [_, x, y, z] <- map (read @Int) . words <$> getLine
  putStrLn $ if (min x y, max x y) `contains` z then "Yes" else "No"
